uuid16_darwin.go

```diff
@@ -0,0 +1,15 @@
+package bluetooth
+
+// New16BitUUID returns a new 128-bit UUID based on a 16-bit UUID.
+//
+// Note: only use registered UUIDs. See
+// https://www.bluetooth.com/specifications/gatt/services/ for a list.
+func New16BitUUID(shortUUID uint16) UUID {
+	// mac OS uses a unique format for UUID.
+	var uuid UUID
+	uuid[0] = 0x00000000
+	uuid[1] = 0x00000000
+	uuid[2] = 0x00000000
+	uuid[3] = uint32(shortUUID) << 16
```

Comment:
If MacOS uses this nonstandard format, then :shrug: 
It's a bit weird though, as the Bluetooth specification has clearly specified how to encode 16-bit UUIDs in a 128-bit UUID. Other platforms use that standard.

Reply:
This is what is returned by cbgo, hence why I had to make these changes. 

macOS does some various other things with anonymizing BLE info, but beats me why this is the case.